From 0901744bb197f6d9fb6d8d120913dd7716bf8301 Mon Sep 17 00:00:00 2001
From: Niu Yawei <yawei.niu@intel.com>
Date: Fri, 12 Jul 2019 08:07:34 +0800
Subject: [PATCH] thread: Export spdk_set_thread()

Export the spdk_set_thread(), so that apps may use it to associate
a thread with SPDK thread when necessary.

Change-Id: I488bea639729d68d7261a497085bbec274357224
Signed-off-by: Niu Yawei <yawei.niu@intel.com>
Reviewed-on: https://review.gerrithub.io/c/spdk/spdk/+/461509
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
Reviewed-by: Changpeng Liu <changpeng.liu@intel.com>
Tested-by: SPDK CI Jenkins <sys_sgci@intel.com>
---
 include/spdk/thread.h          | 7 +++++++
 include/spdk_internal/thread.h | 6 ------
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/include/spdk/thread.h b/include/spdk/thread.h
index 466cbf5b57..25a402c77a 100644
--- a/include/spdk/thread.h
+++ b/include/spdk/thread.h
@@ -205,6 +205,13 @@ void spdk_thread_lib_fini(void);
  */
 struct spdk_thread *spdk_thread_create(const char *name, struct spdk_cpuset *cpumask);
 
+/**
+ * Force the current system thread to act as if executing the given SPDK thread.
+ *
+ * \param thread The thread to set.
+ */
+void spdk_set_thread(struct spdk_thread *thread);
+
 /**
  * Release any resources related to the given thread and destroy it. Execution
  * continues on the current system thread after returning.
diff --git a/include/spdk_internal/thread.h b/include/spdk_internal/thread.h
index fe70b4876a..220d4338c0 100644
--- a/include/spdk_internal/thread.h
+++ b/include/spdk_internal/thread.h
@@ -38,10 +38,4 @@
 
 struct spdk_thread;
 
-/**
- * Force the current system thread to act as if executing the given SPDK thread.
- * This is only used for unit testing.
- */
-void spdk_set_thread(struct spdk_thread *thread);
-
 #endif /* SPDK_THREAD_INTERNAL_H_ */
